pageserver/compaction/src/identify_levels.rs

```diff
@@ -135,6 +135,7 @@ where
             // Is it small enough to be considered part of this level?
             if r.end.0 - r.start.0 > lsn_max_size {
                 // Too large, this layer belongs to next level. Stop.
+                // Due to the sorting bug pointed out above there could still be smaller layers at same key range
```

Comment:
An important point to notice here is that when we see a too large layer and bail out of the loop, we *discard* the candidate that we were building, and return with the "current best" safe stopping point that we had seen earlier.